ee/clickhouse/views/experiments.py

```diff
@@ -0,0 +1,142 @@
+from typing import Any
+
+from rest_framework import request, serializers, viewsets
+from rest_framework.decorators import action
+from rest_framework.exceptions import ValidationError
+from rest_framework.permissions import IsAuthenticated
+from rest_framework.request import Request
+from rest_framework.response import Response
+
+from ee.clickhouse.queries.experiments.funnel_experiment_result import ClickhouseFunnelExperimentResult
+from posthog.api.routing import StructuredViewSetMixin
+from posthog.models.experiment import Experiment
+from posthog.models.feature_flag import FeatureFlag
+from posthog.models.filters.filter import Filter
+from posthog.models.team import Team
+from posthog.permissions import ProjectMembershipNecessaryPermissions, TeamMemberAccessPermission
+
+
+class ExperimentSerializer(serializers.ModelSerializer):
+
+    feature_flag_key = serializers.CharField(source="get_feature_flag_key")
```

Comment:
Q: Why not use a FeatureFlagSerializer instead?

Reply:
I wanted to control what happens to the FF based on how Experiment is being created/updated.

Can I do it using the FFSerializer? (still not quite sure how this works, need to spend more time with Django)

( https://github.com/PostHog/posthog/pull/7492/files#diff-f8f1e9540b8c560bf799086bf6acfc77a2363eeeac79c2ea606b95b0a088fb8fR61-R70 ) &

https://github.com/PostHog/posthog/pull/7492/files#diff-f8f1e9540b8c560bf799086bf6acfc77a2363eeeac79c2ea606b95b0a088fb8fR97-R101

Reply:
Ah. It seems this is used in the creation flow so passing the whole featureflag object in is untenable.

If a case comes up where FE needs full feature flag information we could add another field that will use existing FeatureFlagSerializer.

Reply:
Right, makes sense